hwchase17/langchainjs
# feat: basic unstructured loader for langchainjs

### Summary

Implements a basic Unstructured file loader for langchainjs. The loader posts the specified file to the Unstructured REST API. The API is not public yet, but you can run it locally using the instructions [here](https://github.com/Unstructured-IO/pipeline-general#see-the-unstructured-quick-start-for-the-many-os-dependencies-that-are-required-if-the-ability-to-process-all-file-types-is-desired). The test is disabled temporarily, but once we have a hosted API for the test to hit I can add in a PR to enable it.

Adds `node-fetch` and `form-data` as dependencies. Couldn't figure out how to get the `mixed/multipart-form` post to work without those, but maybe someone who's better than me at node can suggest an alternative :)

### Testing

- Install the Unstructured API locally using the instructions  [here](https://github.com/Unstructured-IO/pipeline-general#see-the-unstructured-quick-start-for-the-many-os-dependencies-that-are-required-if-the-ability-to-process-all-file-types-is-desired).
- Run `make run-web-app`
- Uncomment the code in `langchain/src/document_loaders/tests/unstructured.test.ts`
- Run `yarn test`. The tests should pass and you should see a `200` in the logs for the Unstructured API.
